Improved seo docs page

NeroxTGC opened the pull request:
## Description

> Added Other Pages Meta Tags & Structured data and Schema markup.

## Contributor Checklist

> Make sure to do the following steps if they are applicable to your PR:

- [x] **Update e2e tests**: If you changed the [/template/app](/template/app), then make sure to do any neccessary updates to [/template/e2e-tests](/template/e2e-tests) also.
- [x] **Update demo app**: If you changed the [/template/app](/template/app), then make sure to do any neccessary updates to [/opensaas-sh/app_diff](/opensaas-sh/app_diff) also. Check [/opensaas-sh/README.md](/opensaas-sh/README.md) for details.
- [x] **Update docs**: If needed, update the [/opensaas-sh/blog/src/content/docs](/opensaas-sh/blog/src/content/docs).

NeroxTGC committed:
Update seo.md

Added Other Pages Meta Tags & Structured data and Schema markup

NeroxTGC committed:
Update seo.md

NeroxTGC committed:
Update seo.md

Added Other Pages Meta Tags & Structured data and Schema markup

NeroxTGC committed:
Merge pull request #1 from NeroxTGC/NeroxTGC-patch-1

Update seo.md

Martinsos reviewed (commented):
Awesome work @NeroxTGC , thanks for this!
I think this generaly makes sense and looks good -> I left some comments, but nothing major I believe, let's take care of those and we can merge!

NeroxTGC committed:
Added Robots.txt & missing Alt attributes

NeroxTGC committed:
Update seo.md

NeroxTGC committed:
Merge branch '100-Seo-Landing-Page'

NeroxTGC commented:
@Martinsos I finished updating the document with your suggested changes. If you like, check it one last time.

I've also created a `robots.tsx` default for the template and fixed the missing `alts` on the images to make the Landing Page 100% Seo Score.

Martinsos commented on opensaas-sh/blog/src/content/docs/guides/seo.md:
````diff
@@ -30,7 +30,149 @@ app SaaSTemplate {
+
+```jsx 
+//Add the react-helmet-async import
+import {HelmetProvider } from 'react-helmet-async';
````
```suggestion
import { HelmetProvider } from 'react-helmet-async';
```

Martinsos commented on opensaas-sh/blog/src/content/docs/guides/seo.md:
```diff
@@ -30,7 +30,181 @@ app SaaSTemplate {
+export default function App() {
+    return (
+    <HelmetProvider>
+      <>
```
@NeroxTGC what about this one?

Martinsos commented on opensaas-sh/blog/src/content/docs/guides/seo.md:
````diff
@@ -30,7 +30,149 @@ app SaaSTemplate {
+}
+```
+
+Now, you can set page-specific meta tags  in your React components.
````
```suggestion
Now, you can set page-specific meta tags in your React components.
```

Martinsos commented on opensaas-sh/blog/src/content/docs/guides/seo.md:
````diff
@@ -30,7 +30,149 @@ app SaaSTemplate {
+
+```jsx {14-22}
+//...
+import { Helmet, HelmetProvider } from 'react-helmet-async';
````
redundant import of helmet provider

Martinsos commented on opensaas-sh/blog/src/content/docs/guides/seo.md:
```diff
@@ -30,7 +30,181 @@ app SaaSTemplate {
+
+To ensure that they are valid:
+- [Google rich results test](https://search.google.com/test/rich-results)
+- [Schema validator](https://validator.schema.org/)
```
@NeroxTGC ping on this - what do you think?

Martinsos commented on template/app/public/robots.txt:
```diff
@@ -0,0 +1,6 @@
+Allow: /
+
+Disallow: /admin/
+Disallow: /api/
+Disallow: /auth/
```
hm so api and auth here - do we need that? because those are parts of the server, and server is not hosted at the same url as client, at least not usually.

NeroxTGC replied:
updated, you are right.

Martinsos reviewed (changes requested):
@NeroxTGC nice, there is still some stuff to fix though, I commented!

NeroxTGC committed:
Update opensaas-sh/blog/src/content/docs/guides/seo.md

Co-authored-by: Martin Šošić <Martinsos@users.noreply.github.com>

NeroxTGC committed:
Update opensaas-sh/blog/src/content/docs/guides/seo.md

Co-authored-by: Martin Šošić <Martinsos@users.noreply.github.com>

NeroxTGC committed:
updated Seo guide

NeroxTGC commented:
### Seo Guide

- Removed the extra `<>`.
- Removed redundant `HelmetProvider`.
- Shortened the section on structured data.

---
### OpenSaaS template

- Removed `/api/` &  `/auth/` from `robots.txt`


@Martinsos I don't think I've missed anything now, right?

NeroxTGC committed:
Added lost introduction.
